tangelo/toolboxes/molecular_computation/mm_charges_solver.py

```diff
@@ -0,0 +1,161 @@
+        if file[-3:] != "pdb":
+            mol = openbabel.openbabel.OBMol()
+            conv.SetOutFormat("pdb")
+            conv.ReadFile(mol, file)
+            conv.WriteFile(mol, f"tmp{i}.pdb")
+            files += ["tmp{0}.pdb"]
+        else:
+            files += [file]
+
+    return files
+
+
+def get_default_mm_package() -> Type[MMChargesSolver]:
+    if is_package_installed("rdkit"):
+        return MMChargesSolverOpenMM()
+    elif is_package_installed("openmm"):
+        return MMChargesSolverRDKit()
+    else:
+        return None
+
+
+def get_mm_package(package: str = "default"):
+    """Return the MMCharges instance
+
+    Args:
+        package (str): The MM backend to use
+
+    Returns:
+        Type[MMChargesSolver]: The MMChargesSolver that computes the partial charges for given geometries
+    """
+    if package == "default":
+        return get_default_mm_package()
+    elif package.lower() == "openmm":
+        return MMChargesSolverOpenMM()
+    elif package.lower() == "rdkit":
+        return MMChargesSolverRDKit()
+    else:
+        raise ValueError(f"Requested MM package of {package} is not currently supported in Tangelo")
+
+
+class MMChargesSolverOpenMM(MMChargesSolver):
+    def __init__(self, force_field_params=('amber14-all.xml', 'amber14/tip3pfb.xml')):
+        self.force_field_params = force_field_params
+        pass
+
+    def get_charges(self, files) -> List[Tuple[float, Tuple[float, float, float]]]:
+        from openmm.app.pdbfile import PDBFile
+        from openmm.app.forcefield import ForceField
+        from openmm import NonbondedForce
+        from openmm.app import Modeller
+        import openbabel
+
+        pdbfiles = convert_files_to_pdbs(files)
+
+        pdbs = [PDBFile(file) for file in pdbfiles]
+        modeller = Modeller(pdbs[0].topology, pdbs[0].positions)
+        for pdb in pdbs[1:]:
+            modeller.add(pdb.topology, pdb.positions)
+        forcefield = ForceField(*self.force_field_params)
+        system = forcefield.createSystem(modeller.topology)
+        nonbonded = [f for f in system.getForces() if isinstance(f, NonbondedForce)][0]
+        charges = [nonbonded.getParticleParameters(i)[0]._value for i in range(system.getNumParticles())]
+        # Obtain xyz string
+        mol = openbabel.openbabel.OBMol()
+        conv = openbabel.openbabel.OBConversion()
+        conv.SetInAndOutFormats("pdb", "xyz")
+        geom = []
+        for file in files:
+            _ = conv.ReadFile(mol, file)
+            geometry = conv.WriteString(mol)
+            geom += atom_string_to_list(geometry.split("\n", 2)[2:][0])
+        return charges, geom
+
+
+class MMChargesSolverRDKit(MMChargesSolver):
+    def __init__(self, mmffVariant="MMFF94"):
+        self.mmffVariant = mmffVariant
+        pass
+
+    def get_charges(self, files) -> List[Tuple[float, Tuple[float, float, float]]]:
+        import rdkit
+        from rdkit.Chem import AllChem
```

Comment:
why not just combine l143-144 into `import rdkit.Chem as rdkc` and then just using that directly ?

your code seems to use rdkit.Chem everywhere anyway and you're using `AllChem` only once so its not like importing rdkit entirely or importing AllChem from that submodule helps you.

Reply:
Oddly enough. If I try to do rdkc.AllChem, it says that it does not exist... `AttributeError: module 'rdkit.Chem' has no attribute 'AllChem'`